spark3/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -74,26 +71,20 @@ protected BaseProcedure(TableCatalog tableCatalog) {
     return result;
   }
 
-  // we have to parse both namespace and name as they may be quoted
-  protected Identifier toIdentifier(String namespaceAsString, String name) {
-    String[] namespaceParts = parseMultipartIdentifier(namespaceAsString);
-
-    String[] nameParts = parseMultipartIdentifier(name);
-    Preconditions.checkArgument(nameParts.length == 1, "Name must consist of one part: %s", name);
-
-    return Identifier.of(namespaceParts, nameParts[0]);
-  }
-
-  private String[] parseMultipartIdentifier(String identifierAsString) {
+  private Identifier toIdentifier(String identifierAsString) {
+    CatalogAndIdentifier catalogAndIdentifier;
     try {
-      ParserInterface parser = spark.sessionState().sqlParser();
-      Seq<String> namePartsSeq = parser.parseMultipartIdentifier(identifierAsString);
-      String[] nameParts = new String[namePartsSeq.size()];
-      namePartsSeq.copyToArray(nameParts);
-      return nameParts;
+      catalogAndIdentifier = Spark3Util.catalogAndIdentifier(spark, identifierAsString, tableCatalog);
     } catch (ParseException e) {
-      throw new RuntimeException("Couldn't parse identifier: " + identifierAsString, e);
+      throw new IllegalArgumentException("Cannot parse identifier", e);
```

Comment:
It would be nice to follow this recommendation from the other review: https://github.com/apache/iceberg/pull/1525/files#r538758721

The description could be the argument name.

Reply:
This will require a bit more refactoring since our callers are usually via modifyIcebergTable/withIcebergTable

Reply:
Moved the "toIdentifier" calls into the Procedures and now the modify/with functions take identifiers directly